### PR TITLE
X-Ray Mesons Traitor Item

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Since we're looking to give stealth traitors some more options and I ran into this idea while looking at security cameras, I made goggles which give their user x-ray vision. They're a traitor item disguised as meson goggles. They also have flash resistance for a bit of extra utility.

https://user-images.githubusercontent.com/60792108/154187342-50fa2550-a802-431e-828b-89ffb025dafb.mp4

**Changelog**

:cl: Rane
- add: Added X-Ray meson goggles to the traitor uplink!

### CODE REVIEW

File: Content.Server/XRay/Components/XRayGlassesComponent.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+namespace Content.Server.Xray.Components
+{
+    [RegisterComponent]
+    public sealed class XRayGlassesComponent : Component
+    {
+        [DataField("activationSlot")]
+        public string ActivationSlot = "eyes";
```
Comment (mirrorcult) [inline Content.Server/XRay/Components/XRayGlassesComponent.cs:7]: use SlotFlags

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 2]
```diff
@@ -195,6 +195,14 @@
   description: These protect you from slips while looking like normal sneakers.
   price: 2
 
+- type: uplinkListing
+  id: UplinkClothingEyesXrayMesons
+  category: Armor
+  itemId: ClothingEyesGlassesMesonXRay
+  listingName: x-ray messon goggles
```
Comment (mirrorcult) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:202]:
```suggestion
  listingName: x-ray meson goggles
```

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 3]
```diff
@@ -195,6 +195,14 @@
...
+  description: These let you see everything through walls, and give you flash resistance.
```
Comment (mirrorcult) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:203]:
```suggestion
  description: These let you see everything through walls, and give you flash immunity.
```

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
need some discussion on whether we want this at all, probably

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 4]
```diff
@@ -195,6 +195,14 @@
...
+  price: 5
```
Comment (metalgearsloth) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:204]: If we end up keeping xray goggles then this either needs to be much more expensive, the range needs a nerf, or thermals need to be dropped entirely to balance it.

Comment (Elijahrane) [reply Resources/Prototypes/Catalog/uplink_catalog.yml:204]: I asked on Discord how expensive it should be and this was the most common answer, I can make it more expensive if you want though.

**Comment by Lamrr** [comment]
Could this be adapted to give meson goggles proper functionality - i.e. seeing walls and tiles through walls?

**Comment by metalgearsloth** [comment]
> Could this be adapted to give meson goggles proper functionality - i.e. seeing walls and tiles through walls?

All it does is toggle FOV so you'd need a completely separate implementation.

**Comment by metalgearsloth** [comment]
Sussed it on discord and people seem pretty eh on it for balance reasons. Toggling fov seems like a bad idea (which also includes the genetic mutation if we ever get it).

**Pull request closed without merging**